server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -565,8 +567,18 @@ private void reroute(RoutingAllocation allocation) {
         long rerouteStartTimeNS = System.nanoTime();
         removeDelayMarkers(allocation);
 
+        long startTime = System.nanoTime();
```

Comment:
nit - if the time diff we are looking is in the magnitude of 100s of ms, seconds or minutes, can we explore using System.currentTimeMillis. System.nanos should be used to measure elapsed time with high precision.

Reply:
Applicable for all the places where you are using nanoTime.

Reply:
Removed these logs, which were mostly added for testing

Reply:
It's not about precision, it's more to do with the logical time. Elapsed time should mostly ever be logical time.